app.ts

```diff
@@ -0,0 +1,214 @@
+import { ajax, css } from "jquery";
+
+class RecordManager {
+	firstNumber: number;
+	lastNumber: number;
+	recordCount: number;
+
+	constructor() {
+		this.firstNumber = 0;
+		this.lastNumber = 0;
+		this.recordCount = 0;
+		this.initialize();
+	}
+
+	initialize() {
+		this.createTableHeader();
+		dataManager.fetchRecordCount()
+			.then(count => {
+				this.recordCount = count - 1;
+				this.updateAndDisplayRecords();
+				this.recordEventHandlers();
+				this.handleResize();
+			})
+			.catch(err => {
+				alert('Error fetching and displaying the table records, reload the page');
+				throw err;
+			});
```

Comment:
So... two things.

1. Returning a promise makes catching errors much much easier for the calling function, so i would recommend always doing it. BUT

2. Here you are kicking off two `async` function calls that do not depend on each other, which is beautiful :) This is a great way to speed up an app, but the calling function still needs to be able to handle the error of the function that you are not waiting for. And so now I will be breaking my own rule about not commenting exact code on an onboarding project. This is how you can wait on multiple `async` functions at the same time:

```suggestion
		let p: Promise<void>[] = [];

		p.push(this.createTableHeader());
		let frcp = dataManager.fetchRecordCount()
			.then(count => {
				this.recordCount = count - 1;
				this.updateAndDisplayRecords();
				this.recordEventHandlers();
				this.handleResize();
			})
			.catch(err => {
				alert('Error fetching and displaying the table records, reload the page');
				throw err;
			});
		p.push(frcp);
		
		return Promise.all(p);
```